### ISSUE TITLE
Precompilation failure with regex on Julia nightly macOS

### ISSUE THREAD

**Issue by omus** [JuliaLang/julia#35732]
I am seeing a regular expression compilation error in TimeZones.jl with the Julia nightly on macOS:
```
...
   Building TimeZones → `~/build/JuliaTime/TimeZones.jl/deps/build.log`
ERROR: LoadError: LoadError: LoadError: LoadError: LoadError: InexactError: check_top_bit(Int32, 4294967248)
Stacktrace:
 [1] throw_inexacterror(::Symbol, ::Type{Int32}, ::UInt32) at ./boot.jl:557
 [2] check_top_bit at ./boot.jl:571 [inlined]
 [3] toInt32 at ./boot.jl:620 [inlined]
 [4] Int32 at ./boot.jl:706 [inlined]
 [5] convert at ./number.jl:7 [inlined]
 [6] cconvert at ./essentials.jl:388 [inlined]
 [7] err_message(::UInt32) at ./pcre.jl:149
 [8] jit_compile at ./pcre.jl:132 [inlined]
 [9] compile(::Regex) at ./regex.jl:73
 [10] Regex(::String, ::UInt32, ::UInt32) at ./regex.jl:37
 [11] Regex(::String) at ./regex.jl:60
 [12] @r_str(::LineNumberNode, ::Module, ::Any) at ./regex.jl:109
 [13] #macroexpand#36 at ./expr.jl:108 [inlined]
 [14] macroexpand at ./expr.jl:107 [inlined]
 [15] docm(::LineNumberNode, ::Module, ::Any, ::Any, ::Bool) at ./docs/Docs.jl:521 (repeats 2 times)
 [16] @doc(::LineNumberNode, ::Module, ::String, ::Vararg{Any,N} where N) at ./boot.jl:463
 [17] include(::Function, ::Module, ::String) at ./Base.jl:382
 [18] include at ./Base.jl:370 [inlined]
 [19] include(::String) at /Users/travis/build/JuliaTime/TimeZones.jl/src/tzdata/TZData.jl:1
 [20] top-level scope at /Users/travis/build/JuliaTime/TimeZones.jl/src/tzdata/TZData.jl:28
 [21] include(::Function, ::Module, ::String) at ./Base.jl:382
 [22] include at ./Base.jl:370 [inlined]
 [23] include(::String) at /Users/travis/build/JuliaTime/TimeZones.jl/src/TimeZones.jl:1
 [24] top-level scope at /Users/travis/build/JuliaTime/TimeZones.jl/src/TimeZones.jl:56
 [25] include(::Function, ::Module, ::String) at ./Base.jl:382
 [26] include(::Module, ::String) at ./Base.jl:370
 [27] top-level scope at none:2
 [28] eval at ./boot.jl:331 [inlined]
 [29] eval(::Expr) at ./client.jl:467
 [30] top-level scope at ./none:3
in expression starting at /Users/travis/build/JuliaTime/TimeZones.jl/src/tzdata/compile.jl:173
in expression starting at /Users/travis/build/JuliaTime/TimeZones.jl/src/tzdata/compile.jl:135
in expression starting at /Users/travis/build/JuliaTime/TimeZones.jl/src/tzdata/compile.jl:135
in expression starting at /Users/travis/build/JuliaTime/TimeZones.jl/src/tzdata/TZData.jl:28
in expression starting at /Users/travis/build/JuliaTime/TimeZones.jl/src/TimeZones.jl:56
ERROR: LoadError: Failed to precompile TimeZones [f269a46b-ccf7-5d73-abea-4c690281aa53] to /Users/travis/.julia/compiled/v1.5/TimeZones/2uWP2_9OVWU.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1289
 [3] _require(::Base.PkgId) at ./loading.jl:1029
 [4] require(::Base.PkgId) at ./loading.jl:927
 [5] require(::Module, ::Symbol) at ./loading.jl:922
 [6] include(::String) at ./client.jl:457
 [7] top-level scope at none:5
in expression starting at /Users/travis/build/JuliaTime/TimeZones.jl/deps/build.jl:1
┌ Error: Error building `TimeZones`: 
└ @ Pkg.Operations /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/Pkg/src/Operations.jl:903
...
```
I'll note that 1.5.0-DEV.777 (7583d870a2) is the [first failure](https://travis-ci.org/github/JuliaTime/TimeZones.jl/jobs/682285676) I noticed (the [previous nightly succeeded](https://travis-ci.org/github/JuliaTime/TimeZones.jl/jobs/681945624) 1.5.0-DEV.739 (c93bcd4ad7)). The [latest failure](https://travis-ci.org/github/JuliaTime/TimeZones.jl/jobs/682969512) is 1.5.0-DEV.813 (a1f64488ba).

I've been unable to reproduce the failure locally.

**Comment by omus** [comment]
I suspect that https://github.com/JuliaLang/julia/pull/35607 may be related

**Comment by KristofferC** [comment]
https://github.com/JuliaLang/julia/pull/35728

**Comment by omus** [comment]
Thanks. I just searched through the issues.

**Comment by maleadt** [comment]
That PR will just render the error correctly though, and not fix the underlying cause.

**Comment by JeffBezanson** [comment]
Fixed by #35750
